tests/serializers/test_any.py

```diff
@@ -201,6 +203,55 @@ def test_any_config_timedelta_float_faction():
     assert s.to_json({one_half_s: 'foo'}) == b'{"1.5":"foo"}'
 
 
+@pytest.mark.parametrize('mode', ['float', 'seconds_float'])
+def test_any_config_timedelta_float_negative(mode):
+    s = SchemaSerializer(core_schema.any_schema(), config={'ser_json_timedelta': mode})
+    one_half_s = timedelta(seconds=-1.5)
+    assert s.to_python(one_half_s) == one_half_s
+    assert s.to_python(one_half_s, mode='json') == -1.5
+    assert s.to_json(one_half_s) == b'-1.5'
+
+    assert s.to_python({one_half_s: 'foo'}) == {one_half_s: 'foo'}
+    assert s.to_python({one_half_s: 'foo'}, mode='json') == {'-1.5': 'foo'}
+    assert s.to_json({one_half_s: 'foo'}) == b'{"-1.5":"foo"}'
+
+
+def test_any_config_timedelta_millisecond():
+    s = SchemaSerializer(core_schema.any_schema(), config={'ser_json_timedelta': 'milliseconds_float'})
+    h2 = timedelta(hours=2)
+    assert s.to_python(h2) == h2
+    assert s.to_python(h2, mode='json') == 7200000.0
+    assert s.to_json(h2) == b'7200000.0'
+
+    assert s.to_python({h2: 'foo'}) == {h2: 'foo'}
+    assert s.to_python({h2: 'foo'}, mode='json') == {'7200000': 'foo'}
+    assert s.to_json({h2: 'foo'}) == b'{"7200000":"foo"}'
+
+
+def test_any_config_timedelta_millisecond_fraction():
+    s = SchemaSerializer(core_schema.any_schema(), config={'ser_json_timedelta': 'milliseconds_float'})
+    h2 = timedelta(seconds=1.5)
```

Comment:
Can we maybe add some microseconds here so that we get fractions-of-milliseconds?

Reply:
Yeah, good idea